proxy/proxy.go

```diff
@@ -329,11 +331,42 @@ func sendDirectResponse(w http.ResponseWriter, r *http.Request,
 	// so we can use that.
 	switch {
 	case strings.HasPrefix(r.Header.Get(hdrContentType), hdrTypeGrpc):
-		w.Header().Set("Grpc-Status", strconv.Itoa(int(codes.Internal)))
-		w.Header().Set("Grpc-Message", errInfo)
+		w.Header().Set(hdrGrpcStatus, strconv.Itoa(int(codes.Internal)))
```

Comment:
Any link handy to allow me to better understand whcih headers we're setting here and why?

Reply:
It's not a very good doc, but that's the best I could find: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md

The `Grpc-Status` is the field that holds the numerical value for the gRPC specific states (e.g. `OK`, `INTERNAL`, `UNKNOWN`, `UNIMPLEMENTED`, `UNAVAILABLE` and so on). The `Grpc-Message` field contains the actual error message in case the gRPC status isn't `OK` (0). Those two fields should be sent as a HTTP/2 trailer field and not header field. That's what we tell the golang `http2` library by setting the `Trailer` field with the name of those two fields.

The `Content-Length` and `:status` are probably not necessary as they would be set anyway by the proxy. Maybe I should remove them as they are from trying out stuff.